src/jdk.incubator.foreign/share/classes/jdk/internal/foreign/AbstractNativeScope.java

```diff
@@ -131,4 +135,26 @@ public MemorySegment allocate(long bytesSize, long bytesAlignment) {
             }
         }
     }
+
+    // only for registering
+    private static class EmptyScope extends AbstractNativeScope {
```

Comment:
I tried to replace this with a singleton as well, but realized that doesn't work because then the list of registered segments would be shared. There's also other things like recreating the segment with less access modes that would be nice to re-use the NativeScope implementation for. (though, if needed, it can be re-written specifically for Allocator. But, NativeScope just seems like (almost) the right hammer here).